### PR TITLE
[DOC] [KUBERNETES] Doc for spark-block-cleaner

### PR DESCRIPTION
**Pull request by zwangsheng**
### _Why are the changes needed?_
Add Docs for kyuubi tools spark-block-cleaner.
* Explain the parameters
* Introduction to basic startup
* Give an example

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [X] [Run test](https://kyuubi.readthedocs.io/en/latest/tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by zwangsheng** [comment]
cc @ulysses-you  @yaooqinn

File: docs/tools/index.rst [hunk 1]
```diff
@@ -15,3 +15,4 @@ Develop Tools
     debugging
     community
     developer
+    spark_block_cleaner
```
Comment (ulysses-you) [inline docs/tools/index.rst:18]: I'd like add a new entrance like `tools`, then we can put this into `tools`

File: docs/tools/spark_block_cleaner.md [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+* Read this article
+* An active Kubernetes cluster
+* [Kubectl](https://kubernetes.io/docs/reference/kubectl/overview/)
+
+## Purpose
+
+When running Spark On Kubernetes, we encountered such a situation that after using hostPath volume local-dir, the usage rate of the directory storing shuffle files remained high.
+
+So an additional tool is needed to clear the accumulated block files.
```
Comment (ulysses-you) [inline docs/tools/spark_block_cleaner.md:20]: `clear ` -> `clean up`

File: docs/tools/index.rst (hunk 1 above)
Comment (zwangsheng) [reply docs/tools/index.rst:18]: Good suggestion, i will move this.

File: docs/tools/spark_block_cleaner.md [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+## Usage
+
+### Build Block Cleaner Docker Image
+```shell
+    docker build ./tools/spark-block-cleaner/kubernetes/docker
+```
+
+### Modify spark-block-cleaner.yml
+You need to modify the spark-block-cleaner.yml to fit your current environment.
```
Comment (ulysses-you) [inline docs/tools/spark_block_cleaner.md:30]: **spark-block-cleaner.yml**

File: docs/tools/spark_block_cleaner.md [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+## Related parameters
+
+Name | Default | Meaning
+--- | --- | ---
+CACHE_DIRS | /data/data1,/data/data2 | The target dirs in container path which will clean block files.
+FILE_EXPIRED_TIME | 604800 | Cleaner will clean the block files which current time - last modified time  more than the fileExpiredTime.
+DEEP_CLEAN_FILE_EXPIRED_TIME | 432000 | Deep clean will clean the block files which current time - last modified time  more than the deepCleanFileExpiredTime.
+FREE_SPACE_THRESHOLD | 60 | After first clean, if free Space low than threshold trigger deep clean.
+SCHEDULE_INTERVAL | 3600 | Cleaner sleep between cleaning.
```
Comment (ulysses-you) [inline docs/tools/spark_block_cleaner.md:97]: add the time unit for these config (seconds).

**Comment by ulysses-you** [comment]
@zwangsheng  thank you for working on this, can we put the snapshot about this docs ?

**Comment by zwangsheng** [comment]
POPO20210716-115052
@ulysses-you

File: docs/tools/index.rst [hunk 5]
```diff
@@ -1,17 +1,11 @@
 .. image:: ../imgs/kyuubi_logo.png
    :align: center
 
-Develop Tools
+Tools
```
Comment (yaooqinn) [inline docs/tools/index.rst:4]: Auxiliaries

**Comment by zwangsheng** [comment]
Straighten out the article，cc @ulysses-you  @yaooqinn

File: docs/tools/spark_block_cleaner.md [hunk 6]
```diff
@@ -4,90 +4,105 @@
...
 ## Requirements
 You'd better have cognition upon the following things when you want to use spark-block-cleaner.
 
 * Read this article
 * An active Kubernetes cluster
 * [Kubectl](https://kubernetes.io/docs/reference/kubectl/overview/)
+* [Docker](https://www.docker.com/)
 
 ## Purpose
+During using Spark On Kubernetes with client deploy-mode, we encountered this scenario that the disk responsible for storing Shuffle data accumulates so many files cause the disk overflows.
```
Comment (yaooqinn) [inline docs/tools/spark_block_cleaner.md:20]: Do not tell ‘our’ story, say it as a common case

File: docs/tools/spark_block_cleaner.md [hunk 7]
```diff
@@ -4,90 +4,105 @@
...
+When deploying Spark Block Cleaner, we will configure volumes for the destination folder. Spark Block Cleaner will perceive the folder by the parameter `CACHE_DIRS`. 
+
+Spark Block Cleaner will clear the perceived folder in a fixed loop(which can be configured by `SCHEDULE_INTERVAL`). And Spark Block Cleaner will select folder start with `blockmgr` and `spark` for deletion using the logic Spark uses to create those folders. 
+
+Before deleting those files, Spark Block Cleaner will determine whether it is a recently modified file(depending on whether the file has not been acted on within the specified time which configured by `FILE_EXPIRED_TIME`). Only delete files those beyond that time interval.
+
+And Spark Block Cleaner will check the disk utilization after clean, if the remaining space is less than the specified value(control by `FREE_SPACE_THRESHOLD`), will trigger deep clean(which file expired time control by `DEEP_CLEAN_FILE_EXPIRED_TIME`).
 
 ## Usage
+Before you start using Spark Block Cleaner, you should build its docker images or using official images(TODO).
```
Comment (yaooqinn) [inline docs/tools/spark_block_cleaner.md:40]: ~~or using official images(TODO)~~

File: docs/tools/spark_block_cleaner.md (hunk 6 above)
Comment (zwangsheng) [reply docs/tools/spark_block_cleaner.md:20]: Ok，I will make it common

**Comment by ulysses-you** [comment]
Have you addressed this comment https://github.com/apache/incubator-kyuubi/pull/815#discussion_r670947998 ?

**Comment by zwangsheng** [comment]
> Have you addressed this comment [#815 (comment)](https://github.com/apache/incubator-kyuubi/pull/815#discussion_r670947998) ?

yes, offline communication. It is ok to use `develop tools` and `tools`

**Review by yaooqinn — APPROVED** [review]

**Comment by ulysses-you** [comment]
thanks, merging to master for v1.3.0

**Pull request closed without merging**